### PR TITLE
Policy based authZ for roles

### PR DESCRIPTION
**Pull request by TiagoBrenck**
## Purpose
Refactoring to use policy based authZ since is the current .NET recommendation.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: 5-WebApp-AuthZ/5-1-Roles/README.md [hunk 1]
```diff
@@ -273,15 +273,19 @@ public static IServiceCollection AddMicrosoftIdentityPlatformAuthentication(this
         // [removed for] brevity
 }
 
+/*
 // In code..(Controllers & elsewhere)
-[Authorize(Roles = DirectoryViewers")] // In controllers
+    [Authorize(Roles = DirectoryViewers")] // In controllers
```
Comment (jmprieur) [inline 5-WebApp-AuthZ/5-1-Roles/README.md]: Should it still be Roles? or Policy?

File: 5-WebApp-AuthZ/5-1-Roles/README.md [hunk 2]
```diff
@@ -273,15 +273,19 @@ public static IServiceCollection AddMicrosoftIdentityPlatformAuthentication(this
...
-User.IsInRole("USerReaders"); // In methods
+    User.IsInRole("UserReaders"); // In methods
+*/
+
 ```
 
 ## About the code
 
-1. In the `ConfigureServices` method of `Startup.cs', add the following line:
+1. In the `ConfigureServices` method of `Startup.cs'`, we added the following line to build the `ClaimsIdentity` object using the claims names in the token:
```
Comment (jennyf19) [inline 5-WebApp-AuthZ/5-1-Roles/README.md]: >' [](start = 51, length = 1)

remove the '?

File: 5-WebApp-AuthZ/5-1-Roles/Infrastructure/AppRoles.cs [hunk 3]
```diff
@@ -13,4 +13,10 @@ public static class AppRoles
         public const string UserReaders = "UserReaders";
         public const string DirectoryViewers = "DirectoryViewers";
     }
+
+    public static class AppPolicies
+    {
+        public const string UserReadersOnly = "UserReadersOnly";
```
Comment (jennyf19) [inline 5-WebApp-AuthZ/5-1-Roles/Infrastructure/AppRoles.cs]: nit: not a huge fan of the names, but i also don't have a suggestion at the moment. :/ :)

File: 5-WebApp-AuthZ/5-1-Roles/README.md (hunk 1 above)
Comment (TiagoBrenck) [reply 5-WebApp-AuthZ/5-1-Roles/README.md]: According to Barry Dorrans, the `[Authorize(Roles()]` exists just for backwards compatibility, so I think we would be safe to remove that line of comment and stat pushing the developers to use `[Authorize(Policy()]` instead. 

Fixed

**Pull request merged**